src/sentry/api/endpoints/project_commits.py

```diff
@@ -0,0 +1,35 @@
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.bases.project import ProjectEndpoint, ProjectReleasePermission
+from sentry.api.serializers import serialize
+from sentry.models import Commit
+
+
+class ProjectCommitsEndpoint(ProjectEndpoint):
+    private = True
+    permission_classes = (ProjectReleasePermission,)
+
+    def get(self, request: Request, project) -> Response:
+        """
+        List a Project's Commits
+        `````````````````````````
+
+        Retrieve a list of commits for a given project.
+
+        :pparam string organization_slug: the slug of the organization the
+                                          commit belongs to.
+        :pparam string project_slug: the slug of the project to list the
+                                     commits of.
+        """
+
+        queryset = Commit.objects.filter(
+            organization_id=project.organization_id, releasecommit__project_id=project.id
```

Comment:
Sorry I didn't notice this earlier - I just realised `ReleaseCommit.project_id` is deprecated and we no longer write to it. It's null in all recent rows from what I can see.

What we want to do here it join through `ReleaseCommit` -> `Release` -> `ReleaseProject` to find which projects are associated with a release.

Reply:
Just to clarify, `Commit` -> `ReleaseCommit` -> `Release` -> `ReleaseProject`, right?